compiler/modules/front/resolveTypes.bal

```diff
@@ -329,6 +369,16 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
     panic error("unimplemented type-descriptor");
 }
 
+function nonEmptyType(ModuleSymbols mod, s:ModuleLevelDefn modDefn, s:TypeDesc td, t:SemType semType) returns t:SemType|ResolveTypeError {
+    if !mod.tc.env.isReady() {
+        mod.deferredEmptinessChecks.push({ semType, modDefn, td });
+    }
+    else {
+        check finishEmptinessCheck(mod, semType, modDefn, td);
```

Comment:
finishEmptinessCheck is doing the same thing as nonEmptyType except that it assumes the env is ready and so won't defer.  The naming of the two functions should reflect this and the parameter order should be the same.

e.g. rename finishEmptinessCheck should be something like nonEmptyTypeNoDefer

Reply:
fixed